src/app-layer-detect-proto.c

```diff
@@ -1269,10 +1268,9 @@ static int AppLayerProtoDetectPMMapSignatures(AppLayerProtoDetectPMCtx *ctx)
     int mpm_ret;
     SigIntId id = 0;
 
-    ctx->map = SCMalloc(ctx->max_sig_id * sizeof(AppLayerProtoDetectPMSignature *));
+    ctx->map = SCCalloc(1, ctx->max_sig_id * sizeof(AppLayerProtoDetectPMSignature *));
```

Comment:
Q: Is there any difference b/w
1.
```
SCCalloc(1, ctx->max_sig_id * sizeof(AppLayerProtoDetectPMSignature *));
```
and
2.
```
SCCalloc(ctx->max_sig_id, sizeof(AppLayerProtoDetectPMSignature *));
```
? Why didn't we use (2)?

Reply:
The first argument indicates the number of objects to allocate while the second indicates the object size.

Most of the `SCCalloc` calls allocate for one object so it'd be possible to add an interface, e.g., `SCCallocOne` (e.g.) and replace `SCCalloc(1, object_size)` with `SCCallocOne(object_size)`.

Reply:
I agree with Shivani : `SCCalloc(ctx->max_sig_id, sizeof(AppLayerProtoDetectPMSignature *));` looks nicer

See `git grep 'SCCalloc(1, ' | grep ' \* '` which finds 46 occurences